cmd/sidecars/network-passt-binding/server/server.go

```diff
@@ -153,3 +164,17 @@ func Serve(server *grpc.Server, socket net.Listener, shutdownChan <-chan struct{
 
 	waitForShutdown(server, errChan, shutdownChan)
 }
+
+func primaryPodIfaceNameFromVMI(vmi *vmschema.VirtualMachineInstance) (string, error) {
+	primaryNet := vmispec.LookupPodNetwork(vmi.Spec.Networks)
+	if primaryNet == nil {
+		return "", fmt.Errorf("primary network was not found")
+	}
+
+	ifaceStatus := vmispec.LookupInterfaceStatusByName(vmi.Status.Interfaces, primaryNet.Name)
+	if ifaceStatus == nil {
+		return "", fmt.Errorf("primary network interface status was not found")
+	}
```

Comment:
Those checks already done at NewPasstNetworkConfigurator()
Please move the pod iface name resolution there.
It may require passing around the status ifaces as well.

Reply:
I dont know what your plans, in the context of this PR, I think those check should be done once early as possible because it serve as validation.
The pod iface name is an input, I think it should be addressed early.